crates/subspace-service/src/sync_from_dsn.rs

```diff
@@ -365,19 +365,24 @@ where
             }
         }
 
-        debug!(target: LOG_TARGET, "Finished DSN sync");
+        while notifications.try_next().is_ok() {
```

Comment:
Why was it needed to reorder this? It was at the end to logically show that we discard all the extra notifications that were issues while we were processing the previous notification. I don't see how it fixes or breaks anything being moved here, but I also don't understand why it is needed.

Reply:
I'm not particularly committed to this part of the change.

The block gap was sometimes being hidden by substrate sync, which was one reason the bug was hard to diagnose and fix.

Reducing the amount of time that the atomic is set to "true" in each loop reduces the chance that substrate sync hides any bugs - if there are a lot of sync notifications being sent continuously.

Reply:
I see, so the goal was to move it before atomic specifically. Wasn't obvious, but makes sense now, thanks.